portal/models/user.py

```diff
@@ -304,6 +311,9 @@ class User(db.Model, UserMixin):
         db.Column(db.Integer, default=0, nullable=False)
     last_password_verification_failure = db.Column(db.DateTime, nullable=True)
 
+    # For 2FA
+    otp_secret = db.Column(db.String(16), default=generate_random_secret)
```

Comment:
Naive question: why not use the app's `SECRET_KEY` instead of generating a secret for every user?...

Reply:
security.  without this, you could potentially use a key obtained from one account on a second, provided you hit the time window.

Reply:
Ah, thanks for the explanation- I assumed the tokens had the user_id/email inside the token.